tests-e2e/lists_test.go

```diff
@@ -973,9 +974,35 @@ func TestBumpEventTypesHandling(t *testing.T) {
 	)
 
 	m.MatchResponse(t, bobRes, matchBobSeesRoom2Bumped)
-}
 
-// Test joining a room puts it at the top of your list. What about invites?
+	// The read receipt stuff here specifically checks for the bug in
+	// https://github.com/matrix-org/sliding-sync/issues/83
```

Comment:
I did try making this its own test but couldn't reproduce the behaviour. Having slept on it, I think the crucial thing is that there has to be some other update to the room which updates the LastModifiedTimestamp in the global caches. (Here, Charlie's join.)